cmd/tracee-ebpf/internal/flags/flags-containers.go

```diff
@@ -0,0 +1,106 @@
+package flags
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aquasecurity/tracee/pkg/containers/runtime"
+)
+
+func ContainersHelp() string {
+	return `Select which container runtimes to connect to for container events enrichment.
+By default, if no flag is passed, tracee will automatically detect installed runtimes by going through known runtime socket paths.
+If runtimes are specified, only the ones passed through flags will be connected to through the provided socket file path.
+Supported runtimes are:
+
+1. CRI-O (crio, cri-o)
+2. Containerd (containerd)
+3. Docker (docker)
+
+Example:
+  --containers crio:/var/run/crio/crio.sock
+`
+}
+
+func contains(s []string, val string) bool {
+	for _, str := range s {
+		if str == val {
+			return true
+		}
+	}
+	return false
+}
+
+func PrepareContainers(containerFlags []string) (runtime.Sockets, error) {
+	if len(containerFlags) == 0 {
+		return autoDiscoverSockets(), nil
+	}
+
+	supportedRuntimes := []string{"crio", "cri-o", "containerd", "docker"}
+
+	sockets := runtime.Sockets{}
+
+	for _, flag := range containerFlags {
+		parts := strings.Split(flag, ":")
+		if len(parts) != 2 {
+			return sockets, fmt.Errorf("failed to parse container flags (must be of format {runtime}:{socket path})")
+		}
+		containerRuntime := parts[0]
+
+		if !contains(supportedRuntimes, containerRuntime) {
+			return sockets, fmt.Errorf("provided unsupported container runtime (see --containers help for supported runtimes)")
+		}
+
+		socket := parts[1]
+		err := sockets.Register(runtimeStringToRuntimeId(containerRuntime), socket)
+
+		if err != nil {
+			return sockets, fmt.Errorf("failed to register runtime socket, %s", err.Error())
+		}
+	}
+
+	return sockets, nil
+}
+
+//check default paths for all supported container runtimes and aggregate them
+func autoDiscoverSockets() runtime.Sockets {
+	sockets := runtime.Sockets{}
+	const (
+		defaultContainerd = "/var/run/containerd/containerd.sock"
+		defaultDocker     = "/var/run/docker.sock"
+		defaultCrio       = "/var/run/crio/crio.sock"
+		defaultPodman     = "/var/run/podman/podman.sock"
+	)
+
+	registerSocket(&sockets, "containerd", defaultContainerd)
```

Comment:
Yes, I think adding the autodiscover mechanism to the help is relevant, I'll do that.

Reply:
With regards to debug messages, I went with the opposite way where failures on autodiscovery would be printed to stderr, but maybe we can just print the whole autodiscover process to stdout on debug, errors (which really aren't proper errors) included, WDYT?

Reply:
Id like to have the output in debug. Just like the OSINFO discoveries are printed. And errors should be printed as well. Thanks!

Reply:
Alright, i'm passing debug to the prepare function now - on debug it will print successful and failed autodiscovered socket registers to stdout and stderr respectively. Does that sound reasonable? I think it emulates what the OSInfo does well enough (of course that's all done in main.go unlike this flag).

Reply:
Yep. And add a prefix to the debug message, something like ENRICH: or something you think its meaningful (just like OSInfo: prefix we currently have). Up to you.

Reply:
Done.